compiler/src/dotty/tools/dotc/core/TypeErrors.scala

```diff
@@ -95,7 +97,7 @@ object handleRecursive {
         case _: RecursionOverflow =>
           throw new RecursionOverflow(op, details, exc, weight)
         case _ =>
-          var e = exc
+          var e: Throwable | Null = exc
```

Comment:
Should the `exc` parameter have type `Throwable | Null` instead of widening here?

Reply:
Done

Reply:
`exc` should be non-nullable. If `exc` equals to `null`, we can see `null` will be thrown at the end, which will cause error during runtime definitely. 

The local mutable variable `e` is ok to become `null` when we call `e.getCause`.